src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -593,8 +593,8 @@ std::string JSOneofIndex(const OneofDescriptor* oneof) {
   return StrCat(index);
 }
 
-// Decodes a codepoint in \x0000 -- \xFFFF.
-uint16 DecodeUTF8Codepoint(uint8* bytes, size_t* length) {
+// Decodes a codepoint in \x0000 -- \x10FFFF.
+uint32 DecodeUTF8Codepoint(uint8* bytes, size_t* length) {
```

Comment:
I'm not a JavaScript expert but this seems unlikely to be backwards compatible. Can it be made so?

Reply:
Backwards compatible in what way specifically?

Reply:
This changes the return type. Code that was expecting a uint16 is now getting a uint32 instead. Is that a problem in context? I'm  not sure, but I'm no JavaScript expert.

Reply:
This is C++, not JavaScript. This function is called only from `EscapeJSString`, which is updated accordingly.

Reply:
should be OK then